internal/chain/create_data_account.go

```diff
@@ -20,5 +21,57 @@ func (CreateDataAccount) Validate(st *StateManager, tx *transactions.GenTransact
 		return fmt.Errorf("invalid payload: %v", err)
 	}
 
-	return errors.New("not implemented") // TODO
+	dataAccountUrl, err := url.Parse(body.Url)
+	if err != nil {
+		return fmt.Errorf("invalid account URL: %v", err)
+	}
+
+	//only the ADI can create the data account associated with the ADI
+	if !dataAccountUrl.Identity().Equal(st.SponsorUrl) {
+		return fmt.Errorf("%q cannot sponsor %q", st.SponsorUrl, dataAccountUrl)
+	}
+
+	//if we have a manger book URL, then we need to parse it.
+	var managerBookUrl string
+	if len(body.ManagerKeyBookUrl) != 0 {
+		u, err := url.Parse(body.ManagerKeyBookUrl)
+		if err != nil {
+			return fmt.Errorf("manager key book specified, but url is invalid: %v", err)
+		}
+		managerBookUrl = u.String()
+	}
+
+	//create the data account
+	account := state.NewDataAccount(dataAccountUrl.String(), managerBookUrl)
+
+	//setup key book associated with account
+	if body.KeyBookUrl == "" {
+		account.KeyBook = st.Sponsor.Header().KeyBook
+	} else {
+		keyBookUrl, err := url.Parse(body.KeyBookUrl)
+		if err != nil {
+			return fmt.Errorf("invalid key book URL: %v", err)
+		}
+
+		ssg := new(protocol.KeyBook)
+		err = st.LoadUrlAs(keyBookUrl, ssg)
+		if err != nil {
+			return fmt.Errorf("invalid key book %q: %v", keyBookUrl, err)
+		}
+		copy(account.KeyBook[:], keyBookUrl.ResourceChain())
+	}
+
+	if managerBookUrl != "" {
+		c, err := st.LoadString(managerBookUrl)
+		if err != nil {
+			return fmt.Errorf("invalid key book %q: %v", managerBookUrl, err)
+		}
```

Comment:
If the manager is allowed to belong to another ADI, it might not be present on this BVN

Reply:
I asked paul. if it was it would add a lot of complexity I would think.

Reply:
Yeah. For now, I'm inclined to require that the manager belong to the same ADI. We can always remove the restriction later.

Reply:
We will need to sign from a different ADI to put data in lite data accounts. (to pay the credits).
I believe it will not be so complex once we have the synthetic tx infrastructure in place.

Reply:
i removed the check requiring the manager to be on the same adi.